src/client/tst/IntermittentProducerAutomaticStreamingTest.cpp

```diff
@@ -0,0 +1,192 @@
+#include "ClientTestFixture.h"
+
+using ::testing::WithParamInterface;
+using ::testing::Bool;
+using ::testing::Values;
+using ::testing::Combine;
+
+class IntermittentProducerAutomaticStreamingTest : public ClientTestBase,
+                                    public WithParamInterface< ::std::tuple<UINT64, UINT32> > {
+protected:
+
+    void SetUp() {
+        ClientTestBase::SetUp(false);
+        UINT64 callbackInterval;
+        UINT32 clientInfoVersion;
+        std::tie(callbackInterval, clientInfoVersion) = GetParam();
+        mDeviceInfo.clientInfo.version = CLIENT_INFO_CURRENT_VERSION;
+        mDeviceInfo.clientInfo.callbackPeriod = (UINT64) callbackInterval * HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
+        mDeviceInfo.clientInfo.automaticStreamingFlags = AUTOMATIC_STREAMING_INTERMITTENT_PRODUCER;
+    }
+
+};
+
+TEST_P(IntermittentProducerAutomaticStreamingTest, ValidateTimerInvokedBeforeTime) {
+    // Create new client so param value of callbackPeriod can be applied
+    ASSERT_EQ(STATUS_SUCCESS, CreateClient());
+
+    PKinesisVideoClient client = FROM_CLIENT_HANDLE(mClientHandle);
+
+    EXPECT_EQ((int)client->timerCallbackInvocationCount, 0);
```

Comment:
would recommend swapping the params as the first one is the expected result. Same for the rest of the occurences

Reply:
will do